### PR TITLE
istioctl 1.0.5 (new formula)

### PR DESCRIPTION
**Pull request by murarisumit**
This is related to  https://github.com/istio/istio/issues/10484.

Added command-line utility istioctl.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/istioctl.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Istioctl < Formula
+  desc "Istio configuration command-line utility"
+  homepage "https://istio.io/"
+  url "https://github.com/istio/istio/releases/download/1.0.5/istio-1.0.5-osx.tar.gz"
```
Comment (lembacon) [inline Formula/istioctl.rb]: Formulae in the core tap need to be built from source.

File: Formula/istioctl.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  sha256 "98043349092853796d2eb2709df0bfd029689ffcc70c451dce04ec45f5fabc83"
+
+  def install
+    bin.install "bin/istioctl"
+  end
+
+  test do
+    assert_match "Retrieve policies and rules", shell_output("#{bin}/istioctl get -h")
```
Comment (lembacon) [inline Formula/istioctl.rb]: Please write a test that exercises the real functionality rather than displaying the help text.

**Review by lembacon — CHANGES_REQUESTED** [review]

File: Formula/istioctl.rb (hunk 2 above)
Comment (murarisumit) [reply Formula/istioctl.rb]: I understand, but needs to have kubernetes cluster running with istio to try any of functionality with it. Not sure how can I test without it, any pointers ?

**Comment by igas** [comment]
@murarisumit I made an attempt before, I could not figure out build part, regarding test - you can do sidecar injection to k8s yml file that should be enough.

**Comment by murarisumit** [comment]
@esnible even I tried yesterday, but it gave .git related error, and moved to different things. While trying it again today. I see this in **makefile**,  but when I download the archive there is no `.git` so can do git related operations.
```
Last 15 lines from /Users/sumit/Library/Logs/Homebrew/istioctl/01.make:
2018-12-19 12:08:13 +0530

make
istioctl

fatal: not a git repository (or any of the parent directories): .git
Makefile:161: *** "TAG cannot be empty".  Stop.
/usr/bin/curl --version
/usr/libexec/java_home --xml --failfast
/usr/local/Homebrew/Library/Homebrew/debrew.rb:11:in `raise'
BuildError: Failed executing: make istioctl
1. raise
```

Makefile:161
```
# If tag not explicitly set in users' .istiorc.mk or command line, default to the git sha.
TAG ?= $(shell git rev-parse --verify HEAD)
ifeq ($(TAG),)
  $(error "TAG cannot be empty")
endif
```
@igas can we point somewhere which can help me solve this.

Brew formula 
```
class Istioctl < Formula
  desc "Istio configuration command-line utility"
  homepage "https://github.com/istio/istio"
  url "https://github.com/istio/istio/archive/1.0.5.tar.gz"
  sha256 "48418c5f27ef61403aea79ea57c502cec426117a5de4db713cd3691e2bf91204"

  depends_on "go" => :build

  def install
    ENV["GOPATH"] = buildpath

    srcpath = buildpath/"src/github.com/istio/istio"
    srcpath.install buildpath.children

    cd srcpath do
      system "make", "istioctl"
    end

  end

  test do
    assert_match "Retrieve policies and rules", shell_output("#{bin}/istioctl get -h")
  end
end
```

**Comment by murarisumit** [comment]
Oops my mistake, so stupid of me. Thanks for pointing out.

**Pull request closed without merging**